align workflow

ttung opened the pull request:
### Description
This is the workflow to align the gisaid data.

Depends on #211, #212

#### Issue
[ch127651](https://app.clubhouse.io/genepi/story/127651/implement-caching-of-alignment-in-nexstrain)

### Test plan
successfully invoked ingest -> transform -> align

align crashes, but that's because I under-provisioned the EC2 instance.  Apparently mafft uses a ton of scratch space.  (I opened #217 to fix that)

EDIT: working [run](https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logsV2:log-groups/log-group/$252Faws$252Fbatch$252Fjob/log-events/aspen-batch-job-definition$252Fdefault$252Ff093afc269174c368b369dbc480793b1)

shanaxel42 commented on workflows/align_gisaid/align.sh:
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+set -Eeuo pipefail
```
This is more just a curiosity question not at all a suggestion, but is there a reason you wrote this workflow in bash instead of python?

ttung replied:
A monolithic python script would not work if we want to pick up updates to aspen.  We could do a python script that starts up another python script for the save portion.  Not sure that'd be significantly simpler in python, but I guess it could plausibly be more readable.

shanaxel42 commented on workflows/align_gisaid/config.yaml:
```diff
@@ -0,0 +1,10 @@
+  - defaults/parameters.yaml
+
+# NOTE: when the number of cpus is modified, it would be prudent to modify workflows/transform_gisaid/transform.sh.
+cores: 48
```
damnnnnnn that's a lot of cores

ttung replied:
Well, we could probably do with fewer cores, but there aren't machines with 384GB of memory with fewer cores.

ttung replied:
ok so it seems 384GB of memory isn't enough to consistently finish the alignment.  I increased the memory requirement so now we are using r5.16xlarge (64 cores, 512GB RAM), but decreased the cores because it's not really that helpful beyond a certain point.  hopefully we can schedule other things onto those machines.

shanaxel42 commented on workflows/transform_gisaid/transform.sh:
```diff
@@ -75,3 +75,16 @@ entity_id=$(/aspen/.venv/bin/python workflows/transform_gisaid/save.py
+    --container-overrides "
+      {
+        \"command\": [\"${ASPEN_GIT_REVSPEC}\", \"workflows/align_gisaid/align.sh\", \"${entity_id}\"],
+        \"vcpus\": 48,
```
Is this a redefinition from the config.yaml above?

ttung replied:
No, it's separate.

shanaxel42 reviewed (approved):
lg with some questions

Commit:
align workflow

### Description
This is the workflow to align the gisaid data.

#### Issue
[ch127651](https://app.clubhouse.io/genepi/story/127651/implement-caching-of-alignment-in-nexstrain)

### Test plan
successfully invoked ingest -> transform -> align

align crashes, but that's because I under-provisioned the EC2 instance.  Apparently mafft uses a ton of scratch space.
